packages/berry-pnp/sources/types.ts

```diff
@@ -6,48 +8,48 @@
 
 export type PackageLocator = {name: string, reference: string} | {name: null, reference: null};
 
-export type PackageInformation = {packageLocation: string, packageDependencies: Map<string, string | [string, string]>};
-export type PackageInformationData = {packageLocation: string, packageDependencies: Array<[string, string | [string, string]]>};
+export type PackageInformation<P extends Path = NativePath> = {packageLocation: P, packageDependencies: Map<string, string | [string, string]>};
+export type PackageInformationData<P extends Path = NativePath> = {packageLocation: P, packageDependencies: Array<[string, string | [string, string]]>};
 
-export type PackageStore = Map<string | null, PackageInformation>;
-export type PackageStoreData = Array<[string | null, PackageInformationData]>;
+export type PackageStore<P extends Path = NativePath> = Map<string | null, PackageInformation<P>>;
+export type PackageStoreData<P extends Path = NativePath> = Array<[string | null, PackageInformationData<P>]>;
 
-export type PackageRegistry = Map<string | null, PackageStore>;
-export type PackageRegistryData = Array<[string | null, PackageStoreData]>;
+export type PackageRegistry<P extends Path = NativePath> = Map<string | null, PackageStore<P>>;
+export type PackageRegistryData<P extends Path = NativePath> = Array<[string | null, PackageStoreData<P>]>;
```

Comment:
Why is `PackageRegistry` templated (with a default on `NativePath`) if the `RuntimeState` only works with portable paths? 🤔

Reply:
The `NativePath` version of `PackageInformation` is used in

https://github.com/yarnpkg/berry/blob/6784070b264a4f5170602681d39a2cdfeb4647c6/packages/berry-pnpify/sources/NodePathResolver.ts#L93

A default of `NativePath` indeeds makes little sense. I'll make most not generic and remove the default generic value for `PackageInformation`.

Reply:
Makes sense 👍 Maybe add a comment explaining that the `@berry/pnp` interface typically use portable paths, but the PnP loader interface itself (`getPackageInformation`) returns native paths.